administrator/components/com_fields/helpers/fields.php

```diff
@@ -170,8 +173,23 @@ public static function getFields ($context, $item = null, $prepareValue = false)
 					{
 						try
 						{
+							/*
+							 *On before field render
+							 *Event allow plugins to modfify the output of the field before it is rendered
+							 */
+                            $dispatcher = JEventDispatcher::getInstance();
+							if($field->value)
+								$dispatcher->trigger('onFieldBeforeRender',array($context,'mustache',&$field));
```

Comment:
I believe the events in the render function are not sufficient enough. Because the end user can override this and define a template in the output parameter of the desired custom field using Mustache syntax and thus it will only be the Mustache engine rendering the layout. So that is why I had to place the events there also.